doctest/parts/doctest_fwd.h

```diff
@@ -743,7 +754,8 @@ struct DOCTEST_INTERFACE AssertData
     // for specific exception-related asserts
     bool        m_threw_as;
     const char* m_exception_type;
-    const char* m_exception_string;
+    String      m_exception_string;
```

Comment:
I think this should be replaced with a union. Otherwise we have to construct a new `Contains` on every comparison (which includes a string copy), after we've destroyed the one we've been passed in the constructor.